File: server/src/main/java/com/example/server/model/Task.java
package com.example.server.model;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Builder
@Table
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    private String name;

    private LocalDate limit_date;

    @CreatedDate
    private LocalDate creation_date;


    private int estimated_duration;

    @Builder.Default
    private int duration = 0;

    @Enumerated(EnumType.STRING)
    private Type type;

    @Enumerated(EnumType.STRING)
    private State state;


}

File: frontend/src/types/Task.ts
export type Task = {
   id: bigint | null,
    name: string,
    limit_date : any,
    creation_date: any,
    estimated_duration: number,
    duration : number,
    state : {
       id: bigint | null,
        state: string
    },
    type:{
       id: bigint | null,
        type: string
    }
}

export function getEmptyTask(): Task {
    return {
        id: null,
        name: "",
        limit_date : null,
        creation_date: null,
        estimated_duration: 0,
        duration : 0,
        state : {
            id: null,
            state: ""
        },
        type:{
            id: null,
            type: ""
        }
    }

}